

include/linux/nodemask.h: In function `__first_unset_node':
include/linux/nodemask.h:250: warning: passing arg 1 of `find_next_zero_bit' discards qualifiers from pointer target type

Maybe some patch in -mm triggered this, but it cannot hurt..

Cc: "David S. Miller" <davem@davemloft.net>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-sparc64-akpm/arch/sparc64/lib/find_bit.c  |    5 +++--
 25-sparc64-akpm/include/asm-sparc64/bitops.h |    3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff -puN arch/sparc64/lib/find_bit.c~sparc64-nodemask-build-fix arch/sparc64/lib/find_bit.c
--- 25-sparc64/arch/sparc64/lib/find_bit.c~sparc64-nodemask-build-fix	2005-01-13 20:38:10.077023472 -0800
+++ 25-sparc64-akpm/arch/sparc64/lib/find_bit.c	2005-01-13 20:45:39.466705856 -0800
@@ -50,9 +50,10 @@ found_middle:
  * on Linus's ALPHA routines, which are pretty portable BTW.
  */
 
-unsigned long find_next_zero_bit(unsigned long *addr, unsigned long size, unsigned long offset)
+unsigned long find_next_zero_bit(const unsigned long *addr,
+			unsigned long size, unsigned long offset)
 {
-	unsigned long *p = addr + (offset >> 6);
+	const unsigned long *p = addr + (offset >> 6);
 	unsigned long result = offset & ~63UL;
 	unsigned long tmp;
 
diff -puN include/asm-sparc64/bitops.h~sparc64-nodemask-build-fix include/asm-sparc64/bitops.h
--- 25-sparc64/include/asm-sparc64/bitops.h~sparc64-nodemask-build-fix	2005-01-13 20:39:33.625322200 -0800
+++ 25-sparc64-akpm/include/asm-sparc64/bitops.h	2005-01-13 20:39:42.862917872 -0800
@@ -223,7 +223,8 @@ extern unsigned long find_next_bit(const
  * on Linus's ALPHA routines, which are pretty portable BTW.
  */
 
-extern unsigned long find_next_zero_bit(unsigned long *, unsigned long, unsigned long);
+extern unsigned long find_next_zero_bit(const unsigned long *,
+					unsigned long, unsigned long);
 
 #define find_first_zero_bit(addr, size) \
         find_next_zero_bit((addr), (size), 0)
_
